Include/cpython/unicodeobject.h

```diff
@@ -142,9 +142,16 @@ typedef struct {
         unsigned int ascii:1;
         /* The object is statically allocated. */
         unsigned int statically_allocated:1;
+        // Does the string embed null characters? Possible values:
+        //   0: No
+        //   1: Yes
+        //   2: Unknown, the string must be scanned
```

Comment:
Would not be better to use 0 for "unknown"? It reduces the risk of bugs: if we forget to initialize it, it will left in valid "unknown" state. It can also slightly improve performance, because explicit initialization can be omitted in `_PyUnicode_New()`.

Reply:
I chose 0 to avoid having to add constants for the different values. 0 and 1 for false and true. Only 2 is special. IMO 0 and 1 are more natural.

There are only 3 functions creating strings: PyUnicode_New(), unicode_subtype_new() and static strings. Strings should not be created differently.

Reply:
Memory allocated for new Python str objects are not initialized, all members must be initialized explicitly (as done by my change).

Reply:
If you would prefer constants, which API would you like?

Reply:
I have no preference. I only proposed an idea in case if you did not consider such option. But if you prefer other option, it is fine to me.